docs/development_tips.md

```diff
@@ -0,0 +1,45 @@
+---
+title: "Tips and Tricks"
+---
+
+This page provides some tips and tricks that may help you as a contributor to Home Assistant. The list here is by no means exhaustive, and if you pick up any additional undocumented tips and tricks, please open a PR to add them here.
+
+## Tips and Tricks
+
+### Keep PRs simple
+
+It's in your best interest to keep your PRs as short and simple as possible. It may feel like extra work on your part to split a change you have in mind into smaller changes, but on the plus side:
+
+- Your code is less likely to get accepted with a bug because they will be easier to spot
+- Your PR reviews will generally happen faster and will be easier to iterate through
+- If you don't do this ahead of time, you may be asked to do so during a review, which is a waste of time for both you and the reviewer.
+
+This often comes up when contributors are "modernizing" an integration that hasn't been touched in a while. There are a lot of features that HA has added over the years, and it may be tempting to add support for all of them in a single PR. The right approach is to break the features down into independent functional changes as best you can and to submit the PRs sequentially.
+
+One strategy for handling sequential PRs is to create a branch for the `next` PR off the `current` PR's branch which you can then start writing code against. This is especially useful if you have split up the PRs such that one is dependent on the previous one since you are working off of the code that will be in `dev` once the PR is merged. If you end up adding additional commits to the `current` PR because of changes/review feedback, you can rebase your `next` PR's branch and more easily incorporate any merge conflicts. Once your `current` PR has been merged, squash the commits from the `current` PR branch in the `next` PR branch and then rebase on `dev`.
+
+### Test package dependency changes in Home Assistant
+
+When making changes to a package that an integration is dependent on, it may be useful to test the changes in Home Assistant to verify that the changes you've made are working as you intended. Here's a useful thing to know: if a module folder that an integration is dependent on is located in `/config`, Home Assistant will use that module instead of the one from `PyPI`.
```

Comment:
I know we also have https://developers.home-assistant.io/docs/api_lib_index/#trying-your-library-inside-home-assistant -- which is the workflow I use since I read it there.

Is this a variation / an alternative approach to what is suggested there or are there differences?

Reply:
they accomplish the same thing but I find this approach to be easier because you don't have to pass any special args to `hass`, it just automatically gets handled. Truthfully, I didn't know this section existed. Maybe it's worth adding there as an alternate approach?

Reply:
I've never used the module approach myself (install interactively, mainly to keep my Home Assistant environment seperate/clean of others; and thus the whole work environment is disposable).

Reply:
I'll remove this from here and instead link to the doc above. If we want to add this method, it should be put in that doc I think